test/contract-access-controller-integration.test.js

```
@@ -0,0 +1,226 @@
+        const acManifestDag = await ipfs1.object.get(hash)
+        acManifest = JSON.parse(acManifestDag.toJSON().data)
+      })
+
+      it('makes database use the correct access controller', async () => {
+        assert.equal(acManifest.params.contractAddress, db.access.contract._address)
+      })
+
+      it('saves database manifest file locally', async () => {
+        assert.notEqual(dbManifest, null)
+      })
+
+      it('saves access controller manifest file locally', async () => {
+        assert.notEqual(acManifest, null)
+      })
+
+      describe('database manifest', () => {
+        it('has correct name', async () => {
+          assert.equal(dbManifest.name, 'AABB')
+        })
+
+        it('has correct type', async () => {
+          assert.equal(dbManifest.type, 'feed')
+        })
+
+        it('has correct address', async () => {
+          assert.notEqual(dbManifest.accessController, null)
+          assert.equal(dbManifest.accessController.indexOf('/ipfs'), 0)
+        })
+      })
+
+      describe('access controller manifest', () => {
+        it('has correct type', async () => {
+          assert.equal(acManifest.type, 'eth-contract')
+        })
+
+        it('has correct address', async () => {
+          assert.equal(acManifest.params.contractAddress.indexOf('0x'), 0)
+          assert.equal(acManifest.params.contractAddress, contract._address)
+        })
+      })
+
+      describe('access controls', () => {
+        it('granting access enables to write to the database', async () => {
+          let err
+          try {
+            await db.add('hello?')
+            assert.equal('Should not end here', false)
+          } catch (e) {
+            err = e
+          }
+
+          assert.equal(err, `Error: Could not append entry, key "${id1.id}" is not allowed to write to the log`)
+          await db.access.grant('write', id1.id, { from: accounts[0] })
+          await db.add('hello!')
```

We don't need this add here as it gets called below in the try/catch block, right?

---

First one should throw an error, but separated them to different tests :+1: